src/eigen/diag.jl

```diff
@@ -29,7 +29,7 @@ function diagonalise_all_kblocks(eigensolver, ham::Hamiltonian, nev_per_kpoint::
                                  kpoints=ham.basis.kpoints, guess=nothing,
                                  prec_type=PreconditionerTPA, interpolate_kpoints=true,
                                  tol=1e-6, maxiter=200, n_conv_check=nothing)
-    T = eltype(ham)
+    T = Complex{typeof(ham.basis.Ecut)}
```

Comment:
And this is where I find using Ecut dubious. Our current standard is that the lattice fixes the floating point type of the computation. We should stick to that.

Reply:
It's the T in PlaneWaveBasis{T}, which we use in other places. Should we have a `eltype(basis::PlaneWaveBasis{T}) where T = T`?

Reply:
This is what I had at some point and you did not like it ;).

Reply:
Yeah I'm still not a huge fan, esp. because it's not clear whether this is real or complex, but it's maybe the least worse option... I'll put it on the discuss list